Simple payments styling

artpi opened the pull request:
![zrzut ekranu 2017-07-21 o 18 28 28](https://user-images.githubusercontent.com/3775068/28472858-93b8e49e-6e42-11e7-843c-b2c7e15ab12f.png)

Please continue

artpi committed:
simple-payments: implement styles to the block

artpi committed:
some change

lamosty commented on modules/simple-payments/simple-payments.php:
```diff
@@ -94,9 +95,11 @@ function output_shortcode( $data ) {
-	<div class='jetpack-simple-payments-price'>{$data['price']}</div>
-	{$items}
-	<div class='jetpack-simple-payments-button' id='{$data['dom_id']}_button'></div>
+	<div class='jetpack-simple-payments-payment-row'>
```
We should name the classes as `jetapck-simple-payments__payment-row`. At least that's what we do in Calypso and it's quite nice. Can we rework it?

iamtakashi replied:
We should avoid using underscores for front-end staff. https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/#selectors

retrofox replied:
![image](https://user-images.githubusercontent.com/77539/28491439-e71d561c-6ec7-11e7-9dc6-362e60669b33.png)

We should update the cs class names according to this.

iamtakashi commented:
![image](https://user-images.githubusercontent.com/908665/28473721-bc57b8f8-6e3d-11e7-88d3-f3a0e7b07e46.png)

lamosty commented:
Please merge https://github.com/Automattic/jetpack/pull/7495 before this PR since I've refactored class names.

iamtakashi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -94,9 +95,11 @@ function output_shortcode( $data ) {
 <div class='{$data[ 'class' ]} jetpack-simple-payments-wrapper'>
 	<div class='jetpack-simple-payments-title'>{$data['title']}</div>
 	<div class='jetpack-simple-payments-description'>{$data['description']}</div>
```
Can we have `<p>` inside of `div.jetpack-simple-payments-description`? Theme most likely has spacing for `<p>` and that would avoid to have an arbitrary margin for this specific shortcode.

retrofox replied:
59ea83a9dfe3373ad56fb7d3cc457768f949b3dc

iamtakashi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -94,9 +95,11 @@ function output_shortcode( $data ) {
 <div class='{$data[ 'class' ]} jetpack-simple-payments-wrapper'>
 	<div class='jetpack-simple-payments-title'>{$data['title']}</div>
```
Can we have `<strong>` inside of `jetpack-simple-payments-title` ?

iamtakashi replied:
Actually, that^ was a bad idea. Sorry, let make it bold with CSS as it was.

iamtakashi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -94,9 +95,11 @@ function output_shortcode( $data ) {
-	{$items}
-	<div class='jetpack-simple-payments-button' id='{$data['dom_id']}_button'></div>
+	<div class='jetpack-simple-payments-payment-row'>
+		<div class='jetpack-simple-payments-price'>{$data['price']}</div>
```
Can we have also `<p>` inside of `div.jetpack-simple-payments-price` for the same reason?

iamtakashi replied:
Sorry, also `<strong>` please.

iamtakashi replied:
Ignore the `<strong>` suggestion, sorry!

retrofox replied:
59ea83a9dfe3373ad56fb7d3cc457768f949b3dc

retrofox commented:
> Please merge #7495 before this PR since I've refactored class names.

I'll change the class names according to this comment.

Also, we had to rebase it since there were conflicts.

retrofox committed:
simple-payments: small coding improvements in output_shortcode()

retrofox committed:
simple-payments: do not use underscore in css class
https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/#selectors

retrofox committed:
simple-payments: minor improvements defining endpoints

retrofox committed:
simple-payments: use `<p>` tag for description and price
https://github.com/Automattic/jetpack/pull/7500#discussion_r128818969

retrofox committed:
simple-payments: minor improvements

retrofox committed:
simple-payments: add purchase-box container. improve paypal button styles.

retrofox commented on modules/simple-payments/simple-payments.css:
```diff
@@ -16,6 +16,7 @@
 .jetpack-simple-payments-price {
```
Maybe you have some suggestion here @iamtakashi, using flexbox for instance. Feel free to edit this file.

retrofox commented:
**current state**

<img src="https://user-images.githubusercontent.com/77539/28492482-265cec6c-6edb-11e7-8644-6fdd62aa7f41.png" width="400px" />

https://retrofocs.wpsandbox.me/2017/07/22/workshop-planbook-post/

iamtakashi commented on modules/simple-payments/simple-payments.css:
```diff
@@ -1,21 +1,57 @@
-body .jetpack-simple-payments__wrapper div.jetpack-simple-payments__purchase-message,
-body .jetpack-simple-payments__wrapper div.jetpack-simple-payments__purchase-message a {
+/* stronger rule in order to set the text color */
+body .jetpack-simple-payments-wrapper div.jetpack-simple-payments-purchase-message {
```
As per #7495, let's change it to `<p>` instead of `<div>`.

iamtakashi commented on modules/simple-payments/paypal-express-checkout.js:
```diff
@@ -106,10 +114,13 @@ var PaypalExpressCheckout = {
 			style: {
 				label: 'pay',
-				color: 'blue'
+				shape: 'pill',
```
Just realised there is `rect` option for `shape`. I think I prefer the rectangle than the pill that feels a bit old school to me. Let's go with `rect`.

retrofox commented:
![image](https://user-images.githubusercontent.com/77539/28495205-beb56c0a-6f1c-11e7-8c04-d9b555b73f75.png)

iamtakashi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -86,23 +88,26 @@ function parse_shortcode( $attrs, $content = false ) {
+		return "
+			<div class='{$data['class']} ${cssPrefix}-wrapper'>
+				<p class='${cssPrefix}-purchase-message'></p>
+				<div class='${cssPrefix}-title'>{$data['title']}</div>
```
I think it's safer to have `<p>` inside of the `<div>` as well. This will likely add bottom margin to the title, but we actually need it. If the product doesn't have description there wouldn't be spacing between the title and the price.

iamtakashi replied:
<div class='${cssPrefix}-title'><p>{$data['title']}</p></div>

iamtakashi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -86,23 +88,26 @@ function parse_shortcode( $attrs, $content = false ) {
+				<div class='${cssPrefix}-title'>{$data['title']}</div>
+				<p class='${cssPrefix}-description'>{$data['description']}</p>
+				<div class='${cssPrefix}-purchase-box'>
+					<p class='${cssPrefix}-price'>{$data['price']}</p>
```
It's good that we now have `<p>` around the price but can we keep the outer `div` we had?

`<div class='${cssPrefix}-price'><p>{$data['price']}</p></div>`

iamtakashi replied:
<div class='${cssPrefix}-price'><p>{$data['price']}</p></div>

iamtakashi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -86,23 +88,26 @@ function parse_shortcode( $attrs, $content = false ) {
+			<div class='{$data['class']} ${cssPrefix}-wrapper'>
+				<p class='${cssPrefix}-purchase-message'></p>
+				<div class='${cssPrefix}-title'>{$data['title']}</div>
+				<p class='${cssPrefix}-description'>{$data['description']}</p>
```
It's good that we now have `<p>` around the price but can we keep the outer `div` we had?

`<div class='${cssPrefix}-description'><p>{$data['price']}</p></div>`

iamtakashi commented:
Please work on the image part ASAP. The style and markup can be tweaked after.

lamosty commented:
@retrofox  and others: just a note about
> simple-payments: do not use underscore in css class https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/#selectors (8153dc5ea82747ace680ec4e61a66584c6b0577c)

I was not using underscores for naming classes of a single thing (like `comment-form`. I was using it for naming something which belongs to a single thing (like `comment-form__title`, `comment-form__message`). We use the same convention in Calypso. It's something like having a parent `div` with class `.comment-form` and then a child `div` with class `.title` -- but more specific `.comment-form` on parent and `.comment-form__title` on that child.

Not a blocker, just explanation why I used that convention for class naming.

iamtakashi commented:
@lamosty I got it. But the css class name convention in Calypso is irrelevant here. We need to follow the core's guideline.

retrofox committed:
simple-pyaments: wrap all text into `<p>` elems

artpi commented on modules/simple-payments/paypal-express-checkout.js:
```diff
@@ -4,18 +4,26 @@
  * of simple-payments module.
  */
 
+var primaryCssClassName = 'jetpack-simple-payments';
```
You are aware that you introduced globals in the JS?

artpi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -12,6 +12,8 @@ class Jetpack_Simple_Payments {
 
 	static $shortcode = 'simple-payment';
 
+	static $cssClassnamePrefix = 'jetpack-simple-payments';
```
PHP is snake_case

retrofox replied:
:-o thanks

artpi commented on modules/simple-payments/simple-payments.php:
```diff
@@ -86,23 +88,26 @@ function parse_shortcode( $attrs, $content = false ) {
-";
-		return $output;
+
+		return "
```
This will make tabs in site html code as well, it was tabbed to the left so we dont print out unnecessary tabs

retrofox replied:
ic, I thought it was a mistake. Reverting now.

retrofox committed:
simple-payments: move globals as PaypalExpressCheckout properties

retrofox committed:
simple-payments: coding improvementes.

retrofox reviewed (approved):
LGTM. Feel free to merge before of after that #7503.

retrofox committed:
simple-payments: minor code improvements

lamosty commented on modules/simple-payments/simple-payments.php:
```diff
@@ -86,23 +88,26 @@ function parse_shortcode( $attrs, $content = false ) {
-	<div class='jetpack-simple-payments__button' id='{$data['dom_id']}_button'></div>
+
+		return "
+<div class='{$data['class']} ${cssPrefix}-wrapper'>
```
I think adding the class name prefix as a variable is not very helpful. It's harder to read the styling then and I've never seen class naming in a variable. :) We are also not going to change the class name prefix I think -- why would we?

lamosty commented on modules/simple-payments/simple-payments.css:
```diff
@@ -1,21 +1,57 @@
+	margin-top: 0.3em;
+}
+
+.jetpack-simple-payments-price {
```
We could use flexbox (like we do in Calypso Editor previews) or at least clear the floats in the parent `div`.

retrofox committed:
rebase: squash -> "simple-payments: move globals as PaypalExpressCheckout properties"

iamtakashi commented:
The QTY field still doesn't appear in my testing.

lamosty reviewed (changes requested):
![image](https://user-images.githubusercontent.com/4988512/28529693-818eed76-7091-11e7-9e6e-cee1098fc650.png)

artpi commented:
I dont think that is what we want:
It is worse than state from 7e8d9abc37f709434036f2472f1a418d98e976fe
![zrzut ekranu 2017-07-24 o 17 13 31](https://user-images.githubusercontent.com/3775068/28530290-a20483ac-7093-11e7-86c7-8e4e9a2549e5.png)
